gstreamer-codecs.json

```diff
@@ -0,0 +1,15 @@
+{
+    "name": "gst-libav",
```

Comment:
This is inherited from the runtime itself. Its provioded by the `org.freedesktop.Platfrom.html5-codecs` extension. Was it not working in your testing?

Reply:
Is the commit message unclear?

Reply:
whoops, missed it sorry! Would you mind opening an issue either in freedesktop-sdk or gnome-build-meta to further discuss this and/or making fd.o aware about the issue? Ideally we don't want each multimedia app to have to bundle its own copies.

Reply:
There is a problem with updating the gst cache reliably if you replace ffmpeg inside the app, which is that the patch in the runtime's gst-libav has to hardcode the library path where different ffmpegs are looked for - so if the library moves to an app provided path, the runtime's gst-libav doesn't invalidate the cache. Really the patch should be dynamically adding the actually loaded ffmpeg libraries (via dlinfo) to the dependent files in the gst cache, so it changes cache key whenever the linked ffmpeg changes. (see https://gitlab.freedesktop.org/gstreamer/gst-libav/issues/31#note_104881 )

Reply:
Oops, was typing into a not updated github. Sorry for old news. :)

Reply:
The one question before I push this would be whether I can easily change the `org.gnome.Totem.Codecs` extension to be a `/app/lib` extension rather than one for GStreamer codecs.

Reply:
I think you have that already, no? Anything in /app/lib/codecs/lib will be added to the library path because it has add-ld-path: "lib". The /app/lib/codecs/lib/gstreamer-1.0 path is separately moved to the extension and the gst stuff moved into it, but if you eg build a different libavcodec (provided you match the runtime ABI version) into /app/lib/codecs/lib it will be the one linked to in preference of the one from the runtime.

Reply:
relevant [issue][1] about the ffmpeg situation I guess. Thank you @ramcq 

(not sure how github works, but I can't seem to be able to resolve my discussion)